### ISSUE TITLE
The role of SeekMap

### ISSUE THREAD

**Issue by pajatopmr** [google/ExoPlayer#932]
I noticed that in HLS the seek map is not used, which made me curious about what role it plays and how it does it.

**Comment by pajatopmr** [comment]
In case it matters, this question is asked in the context of trying to understand why seeking forward in a live stream (iTV24) has a very long delay proportional to the distance in the seek.  For example in the 2.5 minute stream seeking to near the end can take around two minutes before the video starts.

**Comment by pajatopmr** [comment]
See https://github.com/google/ExoPlayer/pull/848 for more details if necessary.

**Comment by ojw28** [comment]
- SeekMap represents an index (i.e. some form of time->byte mapping) extracted from the media container. For seeking to work in non-chunked media, it's required that a SeekMap be extracted.
- In HLS the container is typically TS or ADTS. These containers do not contain time->byte mapping information. Hence seeking does not work for playback of non-chunked TS or ADTS.
- In HLS media is chunked, meaning it's provided in segments o 5-10 seconds in length. These segments are listed in media playlist m3u8 files. These files provide a way of seeking at a higher level.

So in summary, for HLS it's expected that SeekMap is not used. Information parsed from the media playlist should be used directly for performing seek operations.

**Comment by pajatopmr** [comment]
May you answer all my questions so well!  Nicely done.

**Comment by pajatopmr** [comment]
HLS seeking for VOD and EVENT playlist types works well.  I have been trying to study this code but have yet to find the methods where positions are mapped to chunks.  Where would I find this code?  My expectation is that once I understand this code I will be able to adapt it for live streams.

**Comment by ojw28** [comment]
Look at the use of seekPositionUs [here](https://github.com/google/ExoPlayer/blob/master/library/src/main/java/com/google/android/exoplayer/hls/HlsChunkSource.java#L255).
